jdaviz/configs/imviz/helper.py

```diff
@@ -620,3 +817,14 @@ def link_image_data(app, link_type='pixels', wcs_fallback_scheme='pixels', wcs_u
             app.data_collection.set_links(links_list)
         app.hub.broadcast(SnackbarMessage(
             'Images successfully relinked', color='success', timeout=8000, sender=app))
+
+
+def _next_subset_num(label_prefix, subset_groups):
+    """Assumes ``prefix i`` format.
+    Does not go back an fill in lower but available numbers. This is consistent with Glue.
+    """
+    labels = [sg.label.split(' ')[1] for sg in subset_groups
+              if sg.label.startswith(label_prefix)]
+    if len(labels) == 0:
+        return 1
+    return max(map(int, labels)) + 1
```

Comment:
Why is the max/map necessary?  Shouldn't the subsets be sorted so the last one in the list has the highest index, under the assumption that none have been renamed but some may have been deleted?

Reply:
I do not trust that assumption enough. What if glue changes that behavior upstream? `max` is fool-proof (until it is not, lol).